sherpa/utils/akima.py

```diff
@@ -0,0 +1,273 @@
+    mmm = 2.0 * mm - m[0]
+    mp = 2.0 * m[n - 2] - m[n - 3]
+    mpp = 2.0 * mp - m[n - 2]
+
+    m1 = numpy.concatenate(([mmm], [mm], m, [mp], [mpp]))
+
+    dm = numpy.abs(numpy.diff(m1))
+    f1 = dm[2:n + 2]
+    f2 = dm[0:n]
+    f12 = f1 + f2
+
+    ids = numpy.nonzero(f12 > 1e-12 * numpy.max(f12))[0]
+    b = m1[1:n + 1]
+
+    b[ids] = (f1[ids] * m1[ids + 1] + f2[ids] * m1[ids + 2]) / f12[ids]
+    c = (3.0 * m - 2.0 * b[0:n - 1] - b[1:n]) / dx
+    d = (b[0:n - 1] + b[1:n] - 2.0 * m) / dx ** 2
+
+    bins = numpy.digitize(xi, x)
+    bins = numpy.minimum(bins, n - 1) - 1
+
+    #
+    #   Try this to fix bad interpolations below first point
+    #
+
+    bins = numpy.maximum(bins,0)
+
+    bb = bins[0:len(xi)]
+    wj = xi - x[bb]
+
+    return ((wj * d[bb] + c[bb]) * wj + b[bb]) * wj + y[bb]
+
+###############################################################################
+
+def akima(xout, xin, yin, axis=-1, out=None):
+    return interpolate(xin, yin, xout, axis, out)
+
+class Akima:
+
+    def __init__(self, x, y):
+        self.b, self.c, self.d, self.n, self.x, self.y = self.init(x, y)
+        return
+
+    def __call__(self, x_new):
+        xi = numpy.array(x_new, dtype=numpy.float64, copy=True)
+        bins = numpy.digitize(xi, self.x)
+        bins = numpy.minimum(bins, self.n - 1) - 1
+
+        #
+        #   Try this to fix bad interpolations below first point
+        #
+
+        bins = numpy.maximum(bins, 0)
+
+        bb = bins[0:len(xi)]
+        wj = xi - self.x[bb]
+
+        result = ((wj * self.d[bb] + self.c[bb]) * wj + self.b[bb]) * wj
+        result += self.y[bb]
+        return result
+
+    def init(self, x, y, axis=-1):
+        n = len(x)
+        if n < 3:
+            raise ValueError("array too small")
+        if n != y.shape[axis]:
+            raise ValueError("size of x-array must match data shape")
+        dx = numpy.diff(x)
+        if any(dx <= 0.0):
+            raise ValueError("x-axis not valid")
+
+        # Remove extrapolation error test
+        #    if any(xi < x[0]) or any(xi > x[-1]):
+        #        raise ValueError("interpolation x-axis out of bounds")
+
+        x = numpy.array(x, dtype=numpy.float64, copy=True)
+        y = numpy.array(y, dtype=numpy.float64, copy=True)
+
+        m = numpy.diff(y) / dx
+        mm = 2.0 * m[0] - m[1]
+        mmm = 2.0 * mm - m[0]
+        mp = 2.0 * m[n - 2] - m[n - 3]
+        mpp = 2.0 * mp - m[n - 2]
+
+        m1 = numpy.concatenate(([mmm], [mm], m, [mp], [mpp]))
+
+        dm = numpy.abs(numpy.diff(m1))
+        f1 = dm[2:n + 2]
+        f2 = dm[0:n]
+        f12 = f1 + f2
+
+        ids = numpy.nonzero(f12 > 1e-12 * numpy.max(f12))[0]
+        b = m1[1:n + 1]
+
+        b[ids] = (f1[ids] * m1[ids + 1] + f2[ids] * m1[ids + 2]) / f12[ids]
+        c = (3.0 * m - 2.0 * b[0:n - 1] - b[1:n]) / dx
+        d = (b[0:n - 1] + b[1:n] - 2.0 * m) / dx ** 2
+        return b, c, d, n, x, y
+
+
+if '__main__' == __name__:
```

Comment:
Can we have this moved into a test, so it's always run? Or at least some sort of a test suite for the akima code? It is run in at least one of the tests added in this PR, but isn't really "unit tested".